libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java

```diff
@@ -72,7 +70,7 @@ public class Encryptor {
      * @return Initialized cipher.
      */
     public static Cipher getEncryptingCipher(String encryptionKey)
-            throws NoSuchAlgorithmException, InvalidAlgorithmParameterException, InvalidKeyException {
```

Comment:
I traced through the code to find some now unused `NoSuchAlgorithmException` references.

Reply:
Could be it was needed for an older Android API level (23 or 21 perhaps?).

Reply:
Now that the underlying methods of ours are not specifying algorithms, this exception is no longer declared by the methods we're calling.